_posts/2017-04-24-solarized.md

```diff
@@ -0,0 +1,545 @@
+the main difference between both modes of the solarized palette.
+
+Finally, I added the following contents to the script:
+
+```bash
+#i3-style solarized-transparent-${newmode} -o ${HOME}/.config/i3/config --reload
+coffee ${HOME}/scripts/cli.coffee solarized-transparent-${newmode} -o ${HOME}/.config/i3/config --reload
+```
+
+If my pull request gets accepted upstream, I'll then be able to use the normal
+i3-style script instead of the current coffee command.
+
+Here is how my i3 statusbar looks like:
+
+![i3blocks solarized](/assets/images/i3blocks-solarized.jpg)
+
+# [](#feh) feh
+
+> feh is an image viewer that can also be used to set the desktop background
+> image.
+
+To set the background image, simply pass the `--bg-fill` parameter to feh:
+
+```console
+$ feh --bg-fill "${HOME}.config/wallpapers/solarized-mountains-${newmode}.png"
+```
+
+That will also create a file `~/.fehbg`, which can then be reused.
+In my i3 configuration, I added the following line for that purpose:
+
+```
+exec_always --no-startup-id ~/.fehbg
+```
+
+For background image, I took a dark solarized image from [Deviant
+Art](http://9beat7.deviantart.com/art/Solarized-Mountains-530027093):
+
+![solarized background dark](/assets/images/solarized-mountains-dark.png)
+
+Then I modified it with [Gimp](https://www.gimp.org/) to get a light solarized version.
+You normally only need to change the 4 background/foreground colors when
+toggling mode (Accent colors staying the same in both modes). But in his image,
+9beat7 used a darkened version of 4 accent colors to ease the transition between
+accent colors and the background color. Hence, I had to modify these shaded parts into
+their lightened counterpart.
+Here is how I proceeded:
+
+```python
+def get_halfway_color(al, bl):
+    '''Function to calculate the color in-between two HSL colors'''
+    for a,b in zip(al,bl):
+        yield min([a,b])+abs(a-b)/2
+
+# [Hue, Saturation, Lightness
+bg_light = [46, 11, 93]
+gold = [45, 100, 71]
+red = [1, 79, 87]
+purple = [238, 44, 77]
+green = [175, 71, 63]
+
+for color in [gold, red, purple, green]:
+    print(list(get_halfway_color(bg_light, color)))
+
+# Output: 
+# [45.5, 55.5, 82.0]
+# [23.5, 45.0, 90.0]
+# [142.0, 27.5, 85.0]
+# [110.5, 41.0, 78.0]
+```
+
+Why did I use HSL colors instead of RGB ? Because it's a much more intuitive way
+to find colors. After playing with it for a bit, I decided to only change the
+saturation and lightness, which gave the following result.
+
+![solarized background light](/assets/images/solarized-mountains-light.png)
+
+Note that it would have been ideal to work with CIELab colors, but since Gimp
+was showing HSL colors right off the bat, it was just simpler to go this way.
+
+
+# [](#termite) Termite
+
+> After using [urxvt](http://software.schmorp.de/pkg/rxvt-unicode.html) for many
+> years, I switched to Termite, which behaves similarly in many ways, and allow
+> painless configurations like adding true transparency.
```

Comment:
>  I switched to Termite, which behaves similarly in many ways, and allow
+> painless configurations like adding true transparency.

 I switched to Termite, which behaves similarly in many ways and offers painless configuration.